hikari/impl/bot.py

```diff
@@ -75,7 +75,7 @@
 _LOGGER: typing.Final[logging.Logger] = logging.getLogger("hikari.bot")
 
 
-class BotApp(traits.BotAware):
+class BotApp(traits.GatewayBotAware):
```

Comment:
This naming makes it really confusing now

Reply:
what's confusing about it

Reply:
The fact that there are 100 different names for the same things. All this name refactoring doesn't bring anything to the pr, it just confuses naming even more

Reply:
so you're saying BotAware should be renamed to BotApp?🤔

Reply:
Still don't see how there are 100 different names for the same thing here though. With a "bot" just being an autonomous program that's separate from Discord and can interact with their users which can be run by itself without the need to introduce any other components as it'll build them itself in this lib with a gateway bot enabling this kind of functionality over by combining multiple sub-components (shards, rest client and multiple dispatch) and receiving gateway events based on user action while a rest bot enables this functionality with only stateless restful sub-components (rest client and rest server) (with the rest server also handling single dispatch which is a design limitation of the rest only flow) and receiving RESTful webhook events based on user action.